src/main/java/ch/njol/skript/effects/EffChange.java

```diff
@@ -234,25 +236,45 @@ else if (mode == ChangeMode.SET)
 				return false;
 			}
 
-			if (changed instanceof Variable && !changed.isSingle() && mode == ChangeMode.SET) {
-				if (ch instanceof ExprParse) {
-					((ExprParse) ch).flatten = false;
-				} else if (ch instanceof ExpressionList) {
-					for (Expression<?> expression : ((ExpressionList<?>) ch).getExpressions()) {
-						if (expression instanceof ExprParse)
-							((ExprParse) expression).flatten = false;
+			if (changed instanceof Variable<?> variable) {
+				// special ExprParse handling
+				if (!variable.isSingle() && mode == ChangeMode.SET) {
+					if (ch instanceof ExprParse exprParse) {
+						exprParse.flatten = false;
+					} else if (ch instanceof ExpressionList<?> exprList) {
+						for (Expression<?> expr : exprList.getExpressions()) {
+							if (expr instanceof ExprParse exprParse) {
+								exprParse.flatten = false;
+							}
+						}
```

Comment:
note that since this is not recursive, it cannot be relied upon to flatter all the possible exprparses
see https://github.com/SkriptLang/Skript/pull/7850

Reply:
im fine to change it here, or someone can do a separate PR and ill just integrate it

Reply:
Addressed in https://github.com/SkriptLang/Skript/pull/7915